### PR TITLE
Add default path for `GenesisFile` and make path relative to configuration yaml file

### PR DESCRIPTION
**Pull request by Jimbo4350**
Issue
-----------

- #611

- This PR **does not result** in breaking changes to upstream dependencies.

Checklist
---------
- [x] This PR contains all the work required to resolve the linked issue.

- [ ] The work contained has sufficient documentation to describe what it does and how to do it.

- [ ] The work has sufficient tests and/or testing.

- [x] I have committed clear and descriptive commits. Be considerate as somebody else will have to read these.

- [x] I have added the appropriate labels to this PR.

### CODE REVIEW

**Comment by ArturWieczorek** [comment]
Can't build it with nix, got an error:

`$ nix-build -A scripts.staging.node`

> [22 of 27] Compiling Cardano.Wallet.Run ( src/Cardano/Wallet/Run.hs, dist/build/Cardano/Wallet/Run.o )
> src/Cardano/Wallet/Run.hs:20:1: error: [-Wunused-imports, -Werror=unused-imports]
>     The import of 'ConfigYamlFilePath'
>     from module 'Cardano.Config.Types' is redundant
>    |
> 20 | import           Cardano.Config.Types ( ConfigYamlFilePath(..), NodeConfiguration(..)
>    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
...
<< Removed unnecessary output >>
...
> builder for '/nix/store/qmj3a0b31g82zisqw2cgsiqh174hkhfh-cardano-node-1.7.0-lib-cardano-node.drv' failed with exit code 1
> cannot build derivation '/nix/store/q1b1603l17b5c649qd6bq743pjn65229-cardano-node-1.7.0-exe-cardano-node-config.drv': 1 dependencies couldn't be built
> cannot build derivation '/nix/store/34k1xhcyh0phyqxcq81gkxbpm69a45zw-cardano-node-1.7.0-exe-cardano-node.drv': 1 dependencies couldn't be built
> cannot build derivation '/nix/store/k44iyd9g5h3zclkyzlv0fyc50wchccwj-cardano-node-staging.drv': 1 dependencies couldn't be built
> error: build of '/nix/store/k44iyd9g5h3zclkyzlv0fyc50wchccwj-cardano-node-staging.drv' failed

When using cabal to build the error above is marked as a warning:

> [22 of 27] Compiling Cardano.Wallet.Run ( src/Cardano/Wallet/Run.hs, /home/artur/Projects/Jordan/cardano-node/dist-newstyle/build/x86_64-linux/ghc-8.6.5/cardano-node-1.7.0/build/Cardano/Wallet/Run.o )
> 
> src/Cardano/Wallet/Run.hs:20:1: warning: [-Wunused-imports]
>     The import of ‘ConfigYamlFilePath’
>     from module ‘Cardano.Config.Types’ is redundant
>    |
> 20 | import           Cardano.Config.Types ( ConfigYamlFilePath(..), NodeConfiguration(..)
>    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...

**Comment by ArturWieczorek** [comment]
I tried to follow on how to get the local node tip from **Local node queries** of `cardano-node` doc and got:

> $ cabal exec cardano-cli -- get-tip --config configuration/log-config-0.liveview.yaml --socket-path socket/0 
> MuxError {errorType = MuxBearerClosed, errorMsg = "<socket: 11> closed when reading data, waiting on next header True", errorStack = []}

Same for `submit-tx.sh` script when trying to submit a tx succesfully created with .`/scripts/issue-genesis-utxo-expenditure.sh`:

> cardano-node$ scripts/submit-tx.sh genesis_transaction_file.bin 
>  cabal v2-run exe:cardano-cli -- submit-tx --tx genesis_transaction_file.bin --config configuration/log-config-0.liveview.yaml --socket-path socket/0
> Up to date
> TxId: txid: txid:6875a2f9
> MuxError {errorType = MuxBearerClosed, errorMsg = "<socket: 11> closed when reading data, waiting on next header True", errorStack = []}
> 
> cardano-cli: MuxBearerClosed "<socket: 11> closed when reading data, waiting on next header True" at

File: cardano-config/src/Cardano/Config/Types.hs [hunk 1]
```diff
@@ -241,9 +243,19 @@ instance FromJSON YamlSocketPath where
...
+ let genFile = unGenesisFile $ ncGenesisFile nc
+ exists <- doesFileExist genFile
+ case exists of
+   -- Genesis file is an absolute path
+   True -> pure nc
+   -- Genesis file is a relative path (relative to configuration yaml filepath)
+   False -> do let d = takeDirectory fp
+               pure $ nc { ncGenesisFile = GenesisFile $ d </> genFile }
+
+-- TODO: Make genesisfile relative to configuration file as above.
```
Comment (dcoutts) [inline cardano-config/src/Cardano/Config/Types.hs:258]: I think this is unnecessary. All we need to do is `takeDirectory fp </> genFile`. That will make it relative to the config file's dir if the file is relative, but if it's absolute it'll remain unchanged.

There is no need to do this probing to see if files exist.

**Pull request merged**